lib/Transforms/Scalar/ScalarReplAggregatesHLSL.cpp

```diff
@@ -4764,17 +4782,18 @@ void SROA_Parameter_HLSL::allocateSemanticIndex(
     if (semantic.empty())
       continue;
 
-    unsigned semGroupEnd = i + 1;
-    while (semGroupEnd < endArgIndex &&
-           FlatAnnotationList[semGroupEnd].GetSemanticString() == semantic) {
-      ++semGroupEnd;
-    }
-
     StringRef baseSemName; // The 'FOO' in 'FOO1'.
     uint32_t semIndex;     // The '1' in 'FOO1'
     // Split semName and index.
     Semantic::DecomposeNameAndIndex(semantic, &baseSemName, &semIndex);
 
+    unsigned semGroupEnd = i + 1;
+    while (semGroupEnd < endArgIndex &&
+           FlatAnnotationList[semGroupEnd].GetSemanticString() == ContinuedPseudoSemantic) {
+      FlatAnnotationList[semGroupEnd].SetSemanticString(baseSemName);
```

Comment:
This line is where I fix the '*' back to the proper semantic name (only the base name, the rest of the function handles the indices)